### DISCUSSION TITLE
Documentation for Expression field

### DISCUSSION THREAD

**Discussion by tschaeffer2** [meatpiHQ/wican-fw#262] [Q&A]
I am a beginner regarding the OBD stuff and using the WiCAN. I was able to update the device and to make the most settings, but it took very long to find out how to manage that. One question is still open: Which formats are allowed in the Expression field (e.g. B2/10, B4>2, B1&0x20,...)?
The source code shows that there seems to be a lot of operations possible, but it is very difficult to find that out by reverse engineering. Is there a documentation available or planned to be available? It there an option to find out if the entered syntax is correct?
I guess a documentation for this and for the other settings PID, Index, Start Bit, Bit Length and the dependence of each other would be helpful to others as well.
Thanks a lot in advance!

**Comment by meatpiHQ** [comment]
@tschaeffer2 I recently updated the expression parser to support more operators, but it's still in main branch and not yet released. I plan to properly document it as part of creating a new vehicle profile.

> (e.g. B2/10, B4>2, B1&0x20,...)?

Hex, & and > are not allowed in the current release.

**Reply by tschaeffer2** [reply]
Thanks a lot for the quick response. Is the feature and a documentation planned?

Btw: WiCAN is a very nice hardware with a great SW.

**Reply by meatpiHQ** [reply]
Yup it's planned.
